src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,11 +1,36 @@
-export const TodoFilter = () => (
+import { Filters } from '../../types/Filters';
+
+type Props = {
+  onFilterChange: (value: React.ChangeEvent<HTMLSelectElement>) => void,
+  onQueryChange: (value: React.ChangeEvent<HTMLInputElement>) => void,
+  onQueryClear: () => void,
+  query: string,
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  onFilterChange,
+  onQueryChange,
+  onQueryClear,
+  query,
+}) => (
   <form className="field has-addons">
     <p className="control">
       <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
+        <select
+          data-cy="statusSelect"
+          onChange={onFilterChange}
+        >
+          {(Object.keys(Filters) as Array<keyof typeof Filters>).map(key => (
```

Comment:
You can use `Object.entries` instead to improve readability 
```suggestion
          {Object.entries(ListStyle).forEach(([key, value]) => (
```